### ISSUE TITLE
[Accessibility issue] Focus outline is not fully visible in several areas of the application (MAS issue 251856)

### ISSUE THREAD

**Issue by ckbedwell** [grafana/grafana#73390]
Low vision users will face difficulty in navigating through the controls and will get misled as to which control the present focus is on. If the boundary is not completely and properly visible around any interactive control present on the element.

**Steps to reproduce**:
Go to play.grafana.org. Use your keyboard to navigate to the areas listed below and notice the focus outline is not fully visible.

**Expected Result**
The focus should be fully visible.

- [ ] Breadcrumbs component.
<img width="396" alt="image" src="https://github.com/grafana/grafana/assets/6906380/789e5d9f-5d31-4b98-a10b-a2a77c78a1fd">

- [ ] DashList component.
<img width="232" alt="image" src="https://github.com/grafana/grafana/assets/6906380/f0862f9f-ade5-4dbd-b987-f35adb3b5b4d">

- [ ] Legend labels on charts
<img width="491" alt="image" src="https://github.com/grafana/grafana/assets/6906380/124e9f36-6d8b-4e5a-9630-2c9bb77ae49f">

_There are more areas, I'll update this ticket as I find them._

I've labelled this with the [upcoming WCAG 2.2 success criterion](https://www.w3.org/TR/WCAG22/#focus-appearance) despite it isn't official yet as it serves as a good reference and what we should be aiming for.
